pkg/distributor/distributor.go

```diff
@@ -1109,17 +1118,6 @@ func (d *Distributor) startPushRequest(ctx context.Context, httpgrpcRequestSize
 	return ctx, rs, nil
 }
 
-func (d *Distributor) checkHttpgrpcRequestSize(rs *requestState, httpgrpcRequestSize int64) error {
-	// If httpgrpcRequestSize was already checked, don't check it again.
-	if rs.httpgrpcRequestSize > 0 {
```

Comment:
This check has been lost. Why it's not required anymore?

Reply:
It was moved outside of this function, into https://github.com/grafana/mimir/blob/bf5dd777ad18fdf6de0a5ddc76e11b34180ff2af/pkg/distributor/distributor.go#L1099

Reply:
I'm sorry. I'm still lost. The check here was on `rs.httpgrpcRequestSize` to avoid double adding the size of 1 request to the accumulater. The check done in the line you linked is not on `rs.httpgrpcRequestSize`.

Reply:
I see what you mean now. This check is in fact unnecessary. We only pass valid `httpgrpcRequestSize` from public `StartPushRequest`, which is called exactly once from gRPC handler. So we can't possibly try to check different httpgrpc sizes.

Reply:
I see what you mean and you're right. I think the check `if rs.httpgrpcRequestSize > 0` was introduced to make the code safer against possible future changes. Given it's an extra `if` condition doesn't look a bad idea to keep it to me (given how tricky is the wiring here) but I also don't feel strong about it.